godot-macros/src/derive_godot_class/property/field_export.rs

```diff
@@ -0,0 +1,517 @@
+        }
+
+        if let Some(list_parser) = parser.handle_list("flags")? {
+            return Self::new_flags(list_parser);
+        }
+
+        if parser.handle_alone("flags_2d_render")? {
+            return Ok(Self::Layers {
+                dimension: LayerDimension::_2d,
+                kind: LayerKind::Render,
+            });
+        }
+
+        if parser.handle_alone("flags_2d_physics")? {
+            return Ok(Self::Layers {
+                dimension: LayerDimension::_2d,
+                kind: LayerKind::Physics,
+            });
+        }
+
+        if parser.handle_alone("flags_2d_navigation")? {
+            return Ok(Self::Layers {
+                dimension: LayerDimension::_2d,
+                kind: LayerKind::Navigation,
+            });
+        }
+
+        if parser.handle_alone("flags_3d_render")? {
+            return Ok(Self::Layers {
+                dimension: LayerDimension::_3d,
+                kind: LayerKind::Render,
+            });
+        }
+
+        if parser.handle_alone("flags_3d_physics")? {
+            return Ok(Self::Layers {
+                dimension: LayerDimension::_3d,
+                kind: LayerKind::Physics,
+            });
+        }
+
+        if parser.handle_alone("flags_3d_navigation")? {
+            return Ok(Self::Layers {
+                dimension: LayerDimension::_3d,
+                kind: LayerKind::Navigation,
+            });
+        }
+
+        match parser.handle_any("file") {
+            Some(None) => {
+                return Ok(Self::File {
+                    global: false,
+                    kind: FileKind::File { filter: None },
+                })
+            }
+            Some(Some(kv)) => {
+                return Ok(Self::File {
+                    global: false,
+                    kind: FileKind::File {
+                        filter: Some(kv.expr()?),
+                    },
+                })
+            }
+            None => (),
+        }
+
+        match parser.handle_any("global_file") {
+            Some(None) => {
+                return Ok(Self::File {
+                    global: true,
+                    kind: FileKind::File { filter: None },
+                })
+            }
+            Some(Some(kv)) => {
+                return Ok(Self::File {
+                    global: true,
+                    kind: FileKind::File {
+                        filter: Some(kv.expr()?),
+                    },
+                })
+            }
+            None => (),
+        }
+
+        if parser.handle_alone("dir")? {
+            return Ok(Self::File {
+                global: false,
+                kind: FileKind::Dir,
+            });
+        }
+
+        if parser.handle_alone("global_dir")? {
+            return Ok(Self::File {
+                global: true,
+                kind: FileKind::Dir,
+            });
+        }
+
+        if parser.handle_alone("multiline")? {
+            return Ok(Self::Multiline);
+        }
```

Comment:
i'd rather make the proc-macro api consistent with the user-facing version godot provides in gdscript. which is `@export_multiline`. because unless you manually do exporting through `get_property_list`, you'll likely not know that godot internally calls this hint `PROPERTY_HINT_MULTILINE_TEXT`.

Reply:
True, in fact I deleted a previous comment after I realized this, but forgot to delete this one as well. All good as-is 👍